http-server-jetty/src/test/groovy/io/micronaut/servlet/jetty/JettyStaticResourceResolutionSpec.groovy

```diff
@@ -15,41 +15,40 @@ import io.micronaut.runtime.server.EmbeddedServer
 import io.micronaut.test.extensions.spock.annotation.MicronautTest
 import io.micronaut.test.support.TestPropertyProvider
 import io.micronaut.web.router.resource.StaticResourceConfiguration
+import spock.lang.Shared
 import spock.lang.Specification
 
 import jakarta.inject.Inject
+import spock.lang.TempDir
+
 import java.nio.file.Paths
-import java.time.Instant
-import java.time.ZoneId
-import java.time.ZonedDateTime
-import java.time.temporal.ChronoUnit
 
 import static io.micronaut.http.HttpHeaders.*
 
 @MicronautTest
 class JettyStaticResourceResolutionSpec extends Specification implements TestPropertyProvider {
 
-    private static File tempFile
+    @TempDir
+    @Shared
+    File tempDir
+
+    @Shared
+    File tempFile
 
-    static {
-        tempFile = File.createTempFile("staticResourceResolutionSpec", ".html")
-        tempFile.write("<html><head></head><body>HTML Page from static file</body></html>")
-        tempFile
+    def setupSpec() {
+        tempFile = File.createTempFile("staticResourceResolutionSpec", ".html", tempDir)
+        tempFile << "<html><head></head><body>HTML Page from static file</body></html>"
     }
 
     @Override
     Map<String, Object> getProperties() {
         [
-                'micronaut.router.static-resources.default.paths': ['classpath:public', 'file:' + tempFile.parent],
+                'micronaut.router.static-resources.default.paths': ['classpath:public', "file:${tempDir.canonicalPath}"],
```

Comment:
This is basically the change (canonical path), the rest of it is using Spock instead of static directory creation